intertidal/exposure.py

```diff
@@ -80,31 +389,163 @@ def exposure(
     # exposure statistics
-    tide_percentiles = np.linspace(0, 1, 101)     
-    
-    # Run the `pixel_tides_ensemble` function with the `calculate_quantiles`
-    # option. For each pixel, an array of tide heights is returned, 
-    # corresponding to the percentiles of the tide range specified by
-    # `tide_percentiles`. 
-    log.info(f"{run_id}: Modelling tide heights for each pixel")
-    tide_cq, _ = pixel_tides_ensemble(
-        ds=dem,
-        ancillary_points="data/raw/tide_correlations_2017-2019.geojson",
+    calculate_quantiles = np.linspace(0, 1, 101)
+
+    # Generate range of times covering entire period of satellite record
+    # for exposure and bias/offset calculation
+    time_range = pd.date_range(
+        start=round_date_strings(start_date, round_type="start"),
+        end=round_date_strings(end_date, round_type="end"),
+        freq=modelled_freq,
+    )
+
+    # Define the temporal filters
+    temp_filters = [
+        "dry",
+        "wet",
+        "summer",
+        "autumn",
+        "winter",
+        "spring",
+        "jan",
+        "feb",
+        "mar",
+        "apr",
+        "may",
+        "jun",
+        "jul",
+        "aug",
+        "sep",
+        "oct",
+        "nov",
+        "dec",
+        "daylight",
+        "night",
+    ]
+
+    # Create empty xarray.Datasets to store outputs into
+    exposure_ds = xr.Dataset(
+        coords=dict(y=(["y"], dem.y.values), x=(["x"], dem.x.values))
+    )
+    modelledtides_ds = xr.Dataset(
+        coords=dict(y=(["y"], dem.y.values), x=(["x"], dem.x.values))
+    )
+
+    # Create an empty dict to store temporal `time_range` variables into
+    timeranges = {}
+
+    # Set filters variable if none supplied
+    if filters is None:
+        filters = ["unfiltered"]
+
+    # If filter combinations are desired, make sure each filter is
+    # calculated individually for later combination
+    if filters_combined is not None:
+        for x in filters_combined:
+            if str(x[0]) not in filters:
+                filters.append(str(x[0]))
+            if str(x[1]) not in filters:
+                filters.append(str(x[1]))
+
+    # Return error for incorrect filter-names
+    all_filters = temp_filters + ["unfiltered"]
+    for x in filters:
+        assert (
+            x in all_filters
+        ), f'Nominated filter "{x}" is not in {all_filters}. Check spelling and retry'
+
+    # Run tide model at low resolution
+    modelledtides_lowres = pixel_tides_ensemble(
+        dem,
         model=tide_model,
+        times=time_range,
         directory=tide_model_dir,
-        calculate_quantiles=tide_percentiles,
-        times=times,
+        ancillary_points="data/raw/tide_correlations_2017-2019.geojson",
+        resample=False,
     )
 
-    # Calculate the tide-height difference between the elevation value and
-    # each percentile value per pixel
-    diff = abs(tide_cq - dem)
+    modelledtides_1d = modelledtides_lowres.mean(dim=["x", "y"])
```

Comment:
There's a similar code block below this that is commented out - can you have a look and see if we can remove one of these?